tiledb/api/src/filter.rs

```diff
@@ -1,345 +1,216 @@
-    ) -> TileDBResult<()> {
-        let c_factor = factor as std::ffi::c_double;
-        self.set_option(
-            ctx,
-            FilterOption::SCALE_FLOAT_FACTOR,
-            &c_factor as *const std::ffi::c_double as *const std::ffi::c_void,
-        )
-    }
-
-    pub fn get_float_factor(&self, ctx: &Context) -> TileDBResult<f64> {
+    pub fn get_float_factor(&self) -> TileDBResult<f64> {
         let mut c_factor: std::ffi::c_double = 0.0;
         self.get_option(
-            ctx,
-            FilterOption::SCALE_FLOAT_FACTOR,
+            ffi::FilterOption::SCALE_FLOAT_FACTOR,
             &mut c_factor as *mut std::ffi::c_double as *mut std::ffi::c_void,
         )
         .map(|_| c_factor as f64)
     }
 
-    pub fn set_float_offset(
-        &self,
-        ctx: &Context,
-        offset: f64,
-    ) -> TileDBResult<()> {
-        let c_offset = offset as std::ffi::c_double;
-        self.set_option(
-            ctx,
-            FilterOption::SCALE_FLOAT_OFFSET,
-            &c_offset as *const std::ffi::c_double as *const std::ffi::c_void,
-        )
-    }
-
-    pub fn get_float_offset(&self, ctx: &Context) -> TileDBResult<f64> {
+    pub fn get_float_offset(&self) -> TileDBResult<f64> {
         let mut c_factor: std::ffi::c_double = 0.0;
         self.get_option(
-            ctx,
-            FilterOption::SCALE_FLOAT_OFFSET,
+            ffi::FilterOption::SCALE_FLOAT_OFFSET,
             &mut c_factor as *mut std::ffi::c_double as *mut std::ffi::c_void,
         )
         .map(|_| c_factor as f64)
     }
 
-    pub fn set_webp_quality(
-        &self,
-        ctx: &Context,
-        quality: f32,
-    ) -> TileDBResult<()> {
-        let c_quality = quality as std::ffi::c_float;
-        self.set_option(
-            ctx,
-            FilterOption::WEBP_QUALITY,
-            &c_quality as *const std::ffi::c_float as *const std::ffi::c_void,
-        )
-    }
-
-    pub fn get_webp_quality(&self, ctx: &Context) -> TileDBResult<f32> {
-        let mut c_factor: std::ffi::c_float = 0.0;
+    pub fn get_positive_delta_max_window(&self) -> TileDBResult<u32> {
+        let mut c_width: std::ffi::c_uint = 0;
         self.get_option(
-            ctx,
-            FilterOption::WEBP_QUALITY,
-            &mut c_factor as *mut std::ffi::c_float as *mut std::ffi::c_void,
-        )
-        .map(|_| c_factor as f32)
-    }
-
-    pub fn set_webp_input_format(
-        &self,
-        ctx: &Context,
-        format: WebPFilterInputFormat,
-    ) -> TileDBResult<()> {
-        let c_format = format as std::ffi::c_uchar;
-        self.set_option(
-            ctx,
-            FilterOption::WEBP_INPUT_FORMAT,
-            &c_format as *const std::ffi::c_uchar as *const std::ffi::c_void,
+            ffi::FilterOption::POSITIVE_DELTA_MAX_WINDOW,
+            &mut c_width as *mut std::ffi::c_uint as *mut std::ffi::c_void,
         )
+        .map(|_| c_width as u32)
     }
 
     pub fn get_webp_input_format(
```

Comment:
This being in the `Filter` impl for all of the filter types does point to this being one of multiple steps to clean this up.

Reply:
Absolutely. I'm sure there's a better way to handle this, but it felt like reaching for `Boxy<dyn Something>` which is beyond where I want to go at this stage.

The only thing that makes me ok with the status quo is knowing that these particular API's will likely only be used by use for implementing `Debug` type traits when folks are just looking for a description of the state.